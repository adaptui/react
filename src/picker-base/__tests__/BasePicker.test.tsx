import * as React from "react";
import { axe, render, press, fireEvent, screen } from "reakit-test-utils";

import {
  PickerBase,
  PickerBaseContent,
  PickerBaseTrigger,
  usePickerBaseState,
  PickerBaseInitialState,
} from "../index";

const PickerBaseComp: React.FC<PickerBaseInitialState> = props => {
  const state = usePickerBaseState({
    ...props,
    pickerId: "picker-1",
    dialogId: "dialog-1",
    baseId: "picker-test",
  });

  return (
    <>
      <PickerBase aria-label="picker base" {...state}>
        <PickerBaseTrigger {...state}>open</PickerBaseTrigger>
      </PickerBase>
      <PickerBaseContent data-testid="picker-content" {...state}>
        Content
      </PickerBaseContent>
    </>
  );
};

describe("PickerBase", () => {
  it("should render correctly", () => {
    const { baseElement } = render(<PickerBaseComp visible={true} />);

    expect(baseElement).toMatchSnapshot();
  });

  it("should open/close properly", () => {
    render(<PickerBaseComp />);

    const pickerContent = screen.getByTestId("picker-content");

    expect(pickerContent).not.toBeVisible();
    fireEvent.click(screen.getByText("open"));
    expect(pickerContent).toBeVisible();

    press.Escape();
    expect(pickerContent).not.toBeVisible();
  });

  test("PickerBase renders with no a11y violations", async () => {
    const { container } = render(<PickerBaseComp />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
