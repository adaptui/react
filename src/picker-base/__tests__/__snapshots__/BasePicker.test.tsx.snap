// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerBase should render correctly 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div>
    <div
      aria-expanded="true"
      aria-haspopup="dialog"
      aria-label="picker base"
      aria-owns="dialog-1"
      id="picker-1"
      role="button"
    >
      <button
        aria-controls="picker-test"
        aria-expanded="true"
        aria-haspopup="dialog"
        tabindex="-1"
        type="button"
      >
        open
      </button>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="__reakit-focus-trap"
    style="position: fixed;"
    tabindex="0"
  />
  <div
    class="__reakit-portal"
  >
    <div
      aria-modal="true"
      data-dialog="true"
      data-testid="picker-content"
      id="dialog-1"
      role="dialog"
      tabindex="-1"
    >
      Content
    </div>
  </div>
  <div
    aria-hidden="true"
    class="__reakit-focus-trap"
    style="position: fixed;"
    tabindex="0"
  />
</body>
`;
